test-data/unit/deps-types.test

```diff
@@ -151,3 +151,411 @@ def h() -> None:
+[file a.py]
+from mod import I
+A = I
+[file mod.py]
+class I: pass
+[out]
+<m.C> -> m.C
+<a.A> -> m
+<mod.I.__init__> -> <m.C.__init__>
+<mod.I> -> m, m.C
+
+[case testAliasDepsGenericMod]
+from mod import I, S, D
+A = D[I, S]
+x: A
+[file mod.py]
+from typing import TypeVar, Generic
+T = TypeVar('T')
+U = TypeVar('U')
+class D(Generic[T, U]): pass
+class I: pass
+class S: pass
+[out]
+<m.A> -> m
+<m.x> -> m
+<mod.D> -> <m.x>, m
+<mod.I> -> <m.x>, m
+<mod.S> -> <m.x>, m
+
+[case testAliasDepsGenericFunc]
+from mod import I, S, D
+A = D[S, I]
+def f(x: A) -> None:
+    pass
+[file mod.py]
+from typing import TypeVar, Generic
+T = TypeVar('T')
+U = TypeVar('U')
+class D(Generic[T, U]): pass
+class I: pass
+class S: pass
+[out]
+<m.A> -> m.f
+<mod.D> -> <m.f>, m, m.f
+<mod.I> -> <m.f>, m, m.f
+<mod.S> -> <m.f>, m, m.f
+
+[case testAliasDepsGenericFuncExtended]
+import a
+def f(x: a.A) -> None:
+    pass
+[file a.py]
+from mod import I, S, D
+A = D[S, I]
+[file mod.py]
+from typing import TypeVar, Generic
+T = TypeVar('T')
+U = TypeVar('U')
+class D(Generic[T, U]): pass
+class I: pass
+class S: pass
+[out]
+<a.A> -> m.f
+<a> -> m
+<mod.D> -> <m.f>, m.f
+<mod.I> -> <m.f>, m.f
+<mod.S> -> <m.f>, m.f
+
+[case testAliasDepsGenericClass]
+from mod import I, D, S, T
+A = D[S, T]
+class C:
+    x: A[I]
+[file mod.py]
+from typing import TypeVar, Generic
+T = TypeVar('T')
+U = TypeVar('U')
+class D(Generic[T, U]): pass
+class I: pass
+class S: pass
+[out]
+<m.A> -> m
+<m.C> -> m.C
+<mod.D> -> <m.C.x>, m
+<mod.I> -> <m.C.x>, m
+<mod.S> -> <m.C.x>, m
+<mod.T> -> m
+
+[case testAliasDepsForwardMod]
+from mod import I
+x: A
+A = I
+[file mod.py]
+from typing import TypeVar, Generic
+class I: pass
+[out]
+<m.A> -> m
+<m.x> -> m
+<mod.I> -> <m.x>, m
+
+[case testAliasDepsForwardFunc]
+from mod import I
+def f(x: A) -> None:
+    pass
+A = I
+[file mod.py]
+class I: pass
+[out]
+<m.A> -> m.f
+<mod.I> -> <m.f>, m, m.f
+
+[case testAliasDepsForwardClass]
+from mod import I
+class C:
+    x: A
+A = I
+[file mod.py]
+class I: pass
+[out]
+<m.A> -> m
+<m.C> -> m.C
+<mod.I> -> <m.C.x>, m
+
+[case testAliasDepsChainedMod]
+from mod import I
+A = I
+B = A
+x: B
+[file mod.py]
+class I: pass
+[out]
+<m.A> -> m
+<m.B> -> m
+<m.x> -> m
+<mod.I> -> <m.x>, m
+
+[case testAliasDepsChainedFunc]
+from mod import I
+A = I
+B = A
+def f(x: B) -> None:
+    pass
+[file mod.py]
+class I: pass
+[out]
+<m.A> -> m
+<m.B> -> m.f
+<mod.I> -> <m.f>, m, m.f
+
+[case testAliasDepsChainedFuncExtended]
+import a
+B = a.A
+def f(x: B) -> None:
+    pass
+[file a.py]
+from mod import I
+A = I
+[file mod.py]
+class I: pass
+[out]
+<m.B> -> m.f
+<a.A> -> m
+<a> -> m
+<mod.I> -> <m.f>, m, m.f
+
+[case testAliasDepsChainedClass]
+from mod import I
+A = I
+B = A
+class C(B):
+    pass
+[file mod.py]
+class I: pass
+[out]
+<m.A> -> m
+<m.B> -> m
+<m.C> -> m.C
+<mod.I.__init__> -> <m.C.__init__>
+<mod.I> -> m, m.C
+
+[case testAliasDepsNestedMod]
+from mod import I, S, D
+A = D[S, I]
+B = D[S, A]
```

Comment:
This test case would be more interesting if `A` and `B` would be defined in different modules, and if we dumped the dependencies within both modules.